test/test_quantized.py

```diff
@@ -1809,6 +1830,27 @@ def test_qnnpack_relu(self, X):
             qY = torch.quantize_per_tensor(Y, scale=scale, zero_point=zero_point, dtype=torch_type)
             self.assertEqual(qY, qY_hat)
 
+    """Tests the correctness of the quantized::qnnpack_tanh op."""
+    @given(X=hu.tensor(shapes=hu.array_shapes(1, 5, 1, 5),
+                       qparams=hu.qparams(dtypes=torch.quint8)))
+    def test_qnnpack_tanh(self, X):
+        # Note: In QNNPACK the output scale and zero_point can be only
+        #       1.0/256, 0 respectively, because it uses an LUT with 256 bins.
+        with override_quantized_engine('qnnpack'):
+            X, (scale, zero_point, torch_type) = X
+
+            X = torch.from_numpy(X)
+            Y = torch.tanh(X)
+
+            qX = torch.quantize_per_tensor(X, scale=scale,
+                                           zero_point=zero_point,
+                                           dtype=torch_type)
+            qY = torch.quantize_per_tensor(Y, scale=1.0 / 128, zero_point=128,
+                                           dtype=torch.quint8)
+            qY_hat = torch.tanh(qX)
+
+            self.assertEqual(qY, qY_hat, message="QNNPACK TanH failed!")
```

Comment:
Should we also compare the results between QNNPACK and server-side tanH implementation?

Reply:
What do you mean? This test is doing exactly that.

Reply:
Yes, I meant server side quantized tanh implementation. Since one of them uses LUT and other does not, I thought it would good to compare them to make sure they don't diverge

Reply:
I don't think there is a point -- if both of them are compared to the FP reference, and both of them are within the tolerance, I am not sure if comparing the two would catch anything that FP ref does not.

But if you think there is still need for it -- it's an easy thing to do

Reply:
Added the tests -- also, rewrote the implementation of the naïve implementation to always have the same output scale/zero_point. That way the accuracies can match across platforms